pkg/controllers/jobset_controller.go

```diff
@@ -264,14 +297,21 @@ func (r *JobSetReconciler) getChildJobs(ctx context.Context, js *jobset.JobSet)
 	return &ownedJobs, nil
 }
 
-func updateReplicatedJobsStatuses(ctx context.Context, js *jobset.JobSet, jobs *childJobs, statuses []jobset.ReplicatedJobStatus) {
+// updateReplicatedJobsStatuses enqueues a JobSet status update to be performed in the status of the
+// replicatedJobs has changed.
+func updateReplicatedJobsStatuses(ctx context.Context, js *jobset.JobSet, statuses []jobset.ReplicatedJobStatus, updateStatusOpts *statusUpdateOpts) {
 	// Check if status ReplicatedJobsStatus has changed
 	if apiequality.Semantic.DeepEqual(js.Status.ReplicatedJobsStatus, statuses) {
 		return
 	}
-	js.Status.ReplicatedJobsStatus = statuses
+	// Add a new status update to perform at the end of the reconciliation attempt.
+	enqueueStatusUpdate(updateStatusOpts, func(jsStatus *jobset.JobSetStatus) {
+		jsStatus.ReplicatedJobsStatus = statuses
```

Comment:
Why not just update js here instead of enqueuing an update? we are doing that for the conditions case, so better follow one pattern.

I suggest we drop the `enqueueStatusUpdate` concept.

Reply:
Oh I forgot to update `updateCondition` to follow this pattern, but to answer your question - I added it to address t [this](https://github.com/kubernetes-sigs/jobset/pull/494#discussion_r1554189775) which suggested we shouldn't compare the oldStatus and new status to determine if an update should be made, but simply track any status updates that should be made. 

The 2 ways I thought of to avoid a deepCopy and deepEqual were either hold a boolean flag in `updateStatusOpts` indicating if status update call should be made, or hold a list of status update functions and only perform the status update call if we have statusUpdateFuncs to run.

In general I tend to prefer other solutions over ones depending on boolean flags but in retrospect I think in this case this alternative is introducing too much extra complexity which is not justified for the amount of value added by it, so I'll refactor to just use a flag.